### DISCUSSION TITLE
SI intro error - wrong subtitles

### DISCUSSION THREAD

**Discussion by Zhelkas** [exult/exult#368] [General]
![SIerror](https://github.com/exult/exult/assets/134815780/b6815fa7-30bd-4b2f-a3d1-10bbe20fc3a4)

Hi everyone,

So I'm playing SI using the latest Exult versions.  Every time I play the intro it displays the wrong subtitles at the wrong time.  I remember it's text from the "into hisss jive" joke intro, yet I never selected anything to show this.

Does anyone know how to fix this so the correct subtitles show when I watch the intro and not this?  Thanks in advance!

**Comment by DominusExult** [comment]
That's very curios. The only way we run the "alternate" intro is by pressing F11 inside the SI game. It shouldn't play when you view the intro from the SI menu.
Which version of Exult are you playing?

**Reply by Zhelkas** [reply]
1.9.0.  I remember this happened in previous versions too.  It occurs both when I play the vanilla game and the SI Fixes mod.

**Reply by DominusExult** [reply]
This is really strange. Can you look for an exultmsg.txt in any of the patch folders and the exult program folder? I can only explain it by somehow an old exultmsg.txt file is being used for overriding this.

**Reply by Zhelkas** [reply]
I looked through all the folders - and couldn't find exultmsg.txt anywhere.  There's textmsg.txt, but that's as close as it gets.

**Reply by DominusExult** [reply]
Where is the textmsg.txt file and what's in it?

**Reply by Zhelkas** [reply]
It's in the serpent/patch folder as well as the mods/Sifixes/patch folder.  The file is "Exult 1.4.03cvs text message file.  Written by ExultStudio."  It has a date of 4/28/2007.

**Reply by DominusExult** [reply]
could you zip the serpent folder and store it somewhere for download and send me the link to forum@exult.info? I need to investigate directly, it seems :)

**Reply by Zhelkas** [reply]
Sure thing.  I sent it to you just now.  Thanks for helping with this.

**Reply by DominusExult** [reply]
just at a quick glance, it is indeed the textmsg.txt file that is messing up things.
Everything is kind of a mess, your patch folder and especially SI Fixes has stuff upon stuff that is likely to break things.
Mixing SI Fixes with the ages old Avatar pack and and and...

I suggest you save your patch folder somewhere so you can return to whatever you did at some later point.
Then also back up your mods folder, create a new one and install SI Fixes anew.

**Reply by Zhelkas** [reply]
Thanks.  I figured it might have been the mess of folders I created earlier.  I'll try cleaning it up and see if that fixes it.  Would SI Fixes work with the Avatar pack, or do you suggest I keep those separate?  I preferred the Avatar Pack version if possible.

**Reply by DominusExult** [reply]
I see that the Avatar pack is the actual culprit. It's textmsg.txt is no longer compatible with Exult. The hex addresses have changed since then and the AP was never updated for that :(

**Reply by Zhelkas** [reply]
Ah, ok.  Well thanks for helping out, now I know what's causing it.  How easy would it be to update the Avatar Pack?  I'm wondering if this is something I can do myself.

**Reply by DominusExult** [reply]
It might be easy but a bit tedious.
Comparing the Avatar Pack textmsg.txt to https://github.com/exult/exult/blob/master/data/exultmsg.txt shows that the addresses are similar but in a different block.
textmsg.txt: 
```
0x110:with help from
0x111:The Exult Team
0x112:Driven by Exult
0x113:The end of Ultima VII
```
exultmsg.txt:
```
0x510:with help from
0x511:The Exult Team
0x512:Driven by Exult
# Ending Messages
0x513:THE END OF ULTIMA VII
```
If you follow that logic (keeping the 0x1.. instead of 0x5..) and adding the blanks we left in should make it work.
This also revealed that the Keyring textmsg.txt is also borked.

**Reply by Zhelkas** [reply]
Thanks, I'll give it a try.  If it doesn't work I'll just keep the Avatar Pack portraits.

**Reply by DominusExult** [reply]
If it works, please share the file, so I can update both Avatar Pack and BG Keyring!

**Comment by Zhelkas** [comment]
> It might be easy but a bit tedious. Comparing the Avatar Pack textmsg.txt to https://github.com/exult/exult/blob/master/data/exultmsg.txt shows that the addresses are similar but in a different block. textmsg.txt:
> 
> ```
> 0x110:with help from
> 0x111:The Exult Team
> 0x112:Driven by Exult
> 0x113:The end of Ultima VII
> ```
> 
> exultmsg.txt:
> 
> ```
> 0x510:with help from
> 0x511:The Exult Team
> 0x512:Driven by Exult
> # Ending Messages
> 0x513:THE END OF ULTIMA VII
> ```
> 
> If you follow that logic (keeping the 0x1.. instead of 0x5..) and adding the blanks we left in should make it work. This also revealed that the Keyring textmsg.txt is also borked.

So I'm trying to figure this out and I'm not sure what you mean here.  textmsg.txt already has the 0x1 addresses.  I don't know what you mean by "adding the blanks we left in".  Am I replacing textmsg.txt with exultmsg.txt?  What am I supposed to be doing here?  Also, when I finish whatever it is I'm supposed to do, should that version go in the "patch" folder or somewhere else?  Thanks.

**Reply by DominusExult** [reply]
See the blanks left in exultmsg.txt https://github.com/exult/exult/blob/3ee5545aa79a00d5378e890f059b5caa96544c01/data/exultmsg.txt#L37

You need to add all the exultmsg.txt lines for the subtitles to the textmsg.txt (only in textmsg the lines begin with 0x1 instead of 0x5).
Then replace your existing textmsg.txt file(s) with that one (mostly wherever you toyed with the Avatar Pack) and the BG Keyring one.

**Reply by Zhelkas** [reply]
That worked!  Thanks so much for your help.  The subtitles are fixed now.

https://www.mediafire.com/file/aha9rm17a6gei8z/textmsg.txt/file

Here's a link to the textmsg.txt file that I modified.  Let me know if it causes any problems.

**Reply by DominusExult** [reply]
Thank you very much. I updated both BG and SI textmsg.txt files in the Avatar Pack and reuploaded the corrected zip.
